From 26764c9463a6e4a236d0f60c11f85a5cb802778a Mon Sep 17 00:00:00 2001
From: Moritz Fischer <moritz.fischer@ettus.com>
Date: Tue, 19 Sep 2017 10:08:33 -0700
Subject: [PATCH 11/13] misc: i2c-eeprom: Deal with different address width

Depending on the eeprom type we'd like to have different addr
widths.

Signed-off-by: Moritz Fischer <moritz.fischer@ettus.com>
---
 drivers/misc/i2c_eeprom.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/drivers/misc/i2c_eeprom.c b/drivers/misc/i2c_eeprom.c
index 9a77c6e164..b0c8e4e231 100644
--- a/drivers/misc/i2c_eeprom.c
+++ b/drivers/misc/i2c_eeprom.c
@@ -61,6 +61,13 @@ static int i2c_eeprom_std_ofdata_to_platdata(struct udevice *dev)
 
 static int i2c_eeprom_std_probe(struct udevice *dev)
 {
+	struct i2c_eeprom *priv = dev_get_priv(dev);
+
+	if (priv->pagewidth == 6)
+		i2c_set_chip_offset_len(dev, 2);
+	else if (priv->pagewidth == 3)
+		i2c_set_chip_offset_len(dev, 1);
+
 	return 0;
 }
 
-- 
2.14.2

